<robot name="whirlybird">

  <!-- materials -->
  <material name="wood">
    <color rgba="0.8 0.7 0 1"/>
  </material>

  <material name="plastic">
    <color rgba="0.5 0.5 0.5 1"/>
  </material>

  <material name="metal">
    <color rgba="0.8 0.8 0.8 1"/>
  </material>

  <!-- links -->
  <link name="base_link">
    <visual>
      <geometry>
        <box size="0.5 0.5 0.01"/>
      </geometry>
      <origin xyz="0 0 0.005" rpy="0 0 0"/>
      <material name="wood"/>
    </visual>
    <visual>
      <geometry>
        <box size="0.2 0.2 0.53"/>
      </geometry>
      <origin xyz="0 0 0.265" rpy="0 0 0"/>
      <material name="wood"/>
    </visual>
  </link>
  
  <link name="gimbal">
    <visual>
      <geometry>
        <box size="0.15 0.15 0.01"/>
      </geometry>
      <origin xyz="0 0 0.005" rpy="0 0 0"/>
      <material name="plastic"/>
    </visual>
    <visual>
      <geometry>
        <box size="0.08 0.01 0.16"/>
      </geometry>
      <origin xyz="0 0.04 0.08" rpy="0 0 0"/>
      <material name="plastic"/>
    </visual>
    <visual>
      <geometry>
        <box size="0.08 0.01 0.16"/>
      </geometry>
      <origin xyz="0 -0.04 0.08" rpy="0 0 0"/>
      <material name="plastic"/>
    </visual>
  </link>

  <link name="pole">
    <visual>
      <geometry>
        <cylinder length="1.4" radius="0.01"/>
      </geometry>
      <origin xyz="0.15 0 0" rpy="0 1.570796 0"/>
      <material name="metal"/>
    </visual>
    <visual>
      <geometry>
        <cylinder length="0.07" radius="0.025"/>
      </geometry>
      <origin xyz="0 0 0" rpy="1.570796 0 0"/>
      <material name="plastic"/>
    </visual>
    <visual>
      <geometry>
        <cylinder length="0.06" radius="0.03"/>
      </geometry>
      <origin xyz="-0.3048 0 0" rpy="0 1.570796 0"/>
      <material name="plastic"/>
    </visual>
  </link>

  <link name="head">
    <visual>
      <geometry>
        <box size="0.1 0.178 0.01"/>
      </geometry>
      <origin xyz="0 0 0.03" rpy="0 0 0"/>
      <material name="plastic"/>
    </visual>
    <visual>
      <geometry>
        <cylinder length="0.02" radius="0.12"/>
      </geometry>
      <origin xyz="0 0.178 0.03" rpy="0 0 0"/>
      <material name="plastic"/>
    </visual>
    <visual>
      <geometry>
        <cylinder length="0.02" radius="0.12"/>
      </geometry>
      <origin xyz="0 -0.178 0.03" rpy="0 0 0"/>
      <material name="plastic"/>
    </visual>
    <visual>
      <geometry>
        <cylinder length="0.08" radius="0.03"/>
      </geometry>
      <origin xyz="0 0 0" rpy="0 1.570796 0"/>
      <material name="plastic"/>
    </visual>
    <visual>
      <geometry>
        <box size="0.08 0.06 0.03"/>
      </geometry>
      <origin xyz="0 0 0.015" rpy="0 0 0"/>
      <material name="plastic"/>
    </visual>
  </link>

  <!-- joints -->
  <joint name="yaw_joint" type="continuous">
    <parent link="base_link"/>
    <child link="gimbal"/>
    <origin xyz="0 0 0.53" rpy="0 0 0"/>
    <axis xyz="0 0 1"/>
  </joint>

  <joint name="pitch_joint" type="continuous">
    <parent link="gimbal"/>
    <child link="pole"/>
    <origin xyz="0 0 0.12" rpy="0 0 0"/>
    <axis xyz="0 1 0"/>
  </joint>

  <joint name="roll_joint" type="continuous">
    <parent link="pole"/>
    <child link="head"/>
    <origin xyz="0.85 0 0" rpy="0 0 0"/>
    <axis xyz="1 0 0"/>
  </joint>
</robot>
